BackEndLib/Wchar.cpp

```diff
@@ -885,3 +893,45 @@ WSTRING WCSReplace(
 		processed = loc + from.size();
 	}
 }
+
+//*****************************************************************************
+WSTRING WCSToLower(WSTRING const &source)
+//Converts WSTRING to lowercase
+{
+	WSTRING lowercased;
+	lowercased.reserve(source.size());
+	for (WSTRING::const_iterator it = source.begin(); it != source.end(); ++it)
+		lowercased += towlower(*it);
+
+	return lowercased;
+}
+
+//*****************************************************************************
+const std::vector<WSTRING> WCSExplode(WSTRING const &source, WCHAR const delimiter)
```

Comment:
Ditto, returning by `const` value is not useful and inhibits move semantics. See https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f20-for-out-output-values-prefer-return-values-to-output-parameters